include/common_structs.h

```diff
@@ -50,6 +55,11 @@ typedef struct Vec2s {
     /* 0x02 */ s16 y;
 } Vec2s; // size = 0x04
 
+typedef struct Vec2su {
```

Comment:
I'm starting to not like our naming system for these vec structs... we use `u16` for short unsigned but then "su" here...

Could we consider a Vec2(T) and Vec3(T) macro (`#define Vec2(T) struct { T x; T y; }`) instead? i.e. `Vec2(u16) pos = { 0, 0 };` (not screaming caps VEC2 because they're effectively type templates)

Reply:
that seems overcomplicated to me, personally, and I could see it making figuring out the actual types and sizes of things confusing. happy to revisit the naming convention for the vec structs though. we already have one Vec(n)(t)u so I just copied that convention for this one

Reply:
We could also typedef them as Vec2u16 to avoid the macro